pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -36,8 +36,11 @@ type NodeAffinity struct {
 	addedPrefSchedTerms *nodeaffinity.PreferredSchedulingTerms
 }
 
+var _ framework.PreFilterPlugin = &NodeAffinity{}
 var _ framework.FilterPlugin = &NodeAffinity{}
+var _ framework.PreScorePlugin = &NodeAffinity{}
 var _ framework.ScorePlugin = &NodeAffinity{}
+var _ framework.ScoreExtensions = &NodeAffinity{}
```

Comment:
Normally, the check is only added for extension point, should us add this to other plugin like this? e.g.
https://github.com/kubernetes/kubernetes/blob/d8f9e4587ac1265efd723bce74ae6a39576f2d58/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L254-L256

Reply:
no need for this one, it is part of ScorePlugin interface.

Reply:
Okey, thanks
revert this change.